azure-iot-device/tests/iothub/test_sync_clients.py

```diff
@@ -40,67 +62,154 @@ def test_from_authentication_provider_instantiates_client(
             pytest.param("MqTt", PipelineAdapter, id="mIxEd CaSe"),
         ],
     )
-    def test_from_authentication_provider_boundary_case_transport_name(
-        self, auth_provider, protocol, expected_pipeline
-    ):
+    def test_boundary_case_protocol_name_format(self, auth_provider, protocol, expected_pipeline):
         client = self.client_class.from_authentication_provider(auth_provider, protocol)
         assert isinstance(client, self.client_class)
         assert isinstance(client._pipeline, expected_pipeline)
 
+    @pytest.mark.it("Raises ValueError on invalid protocol input")
     @pytest.mark.parametrize("auth_provider", ["SymmetricKey"], ids=[""], indirect=True)
-    def test_from_authentication_provider_bad_input_raises_error_transport_name(
-        self, auth_provider
-    ):
+    # TODO: handle the auth_provider fixture differently so it doesn't result in empty [] in output
+    def test_bad_protocol_input_raises_error(self, auth_provider):
         with pytest.raises(ValueError):
             self.client_class.from_authentication_provider(auth_provider, "bad input")
 
-    def test_instantiation_sets_on_connected_handler_in_transport(self, client):
-        assert client._pipeline.on_transport_connected is not None
-        assert client._pipeline.on_transport_connected == client._on_state_change
 
-    def test_instantiation_sets_on_disconnected_handler_in_transport(self, client):
-        assert client._pipeline.on_transport_disconnected is not None
-        assert client._pipeline.on_transport_disconnected == client._on_state_change
+class SharedClientConnectTests(object):
+    @pytest.mark.it("Begins a 'connect' pipeline operation")
+    def test_calls_pipeline_connect(self, client, pipeline):
+        client.connect()
+        assert pipeline.connect.call_count == 1
 
-    def test_instantiation_sets_on_method_request_received_handler_in_transport(self, client):
-        assert client._pipeline.on_transport_method_request_received is not None
-        assert (
-            client._pipeline.on_transport_method_request_received
-            == client._inbox_manager.route_method_request
-        )
+    @pytest.mark.it("Waits for the completion of the 'connect' pipeline operation before returning")
+    def test_waits_for_pipeline_op_completion(self, mocker, client_manual_cb, pipeline_manual_cb):
```

Comment:
>client_manual_cb [](start = 60, length = 16)

how does this even work?  client_manual_cb isn't even defined in this scope as far as I can tell.

Reply:
never mind.  figured it out.

---
In reply to: [291322441](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r291322441) [](ancestors = 291322441)